Update json-and-xml-functors.md to reflect Documentation note for quotes in json functor (#984)
harness/developer-hub: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md

```json
{\n  "data": {\n    "attributes": {\n      "name": "new-construction-api",\n      "version_pins": {\n        "mvn-service://new-construction-api": "0.0.253-feature_NC-6595.8d268cd~nc1.6312a66"\n      }\n    }\n  }\n}\n
```

```
NOTE:

1. JSON accepts the control sequence `\n` as strings. To format JSON, use `jq` to prettify the JSON.

2. Conditional expressions within double quotes are considered strings.
   So, "<+json.select("fields.status.name", httpResponseBody)>"=="In Progress" is treated as string comparison and will not work.
   Use `<+json.select("fields.status.name", httpResponseBody)>=="In Progress"` instead.

>    Similarly, use `<+json.object(httpResponseBody).fields.parent>!=null` for null comparison.

Reviewer: The keyword null, too, shouldn't be enclosed in quotes during comparison. Here's an example of a null comparison: <+json.object(httpResponseBody).fields.parent>!=null

Author: done

```json
{\n  "data": {\n    "attributes": {\n      "name": "new-construction-api",\n      "version_pins": {\n        "mvn-service://new-construction-api": "0.0.253-feature_NC-6595.8d268cd~nc1.6312a66"\n      }\n    }\n  }\n}\n
```

```
NOTE:

> 1. JSON accepts the control sequence `\n` in strings. Use a prettifier like jq to format the JSON.

Reviewer 2: ```suggestion
JSON accepts the control sequence `\n` as strings. To format JSON, use `jq` to prettify the JSON.
```

Author: done

> 2. If you are passing the condition "<+json.object(httpResponseBody).fields.parent>" in double quotes which is considered to be a string.

Reviewer 2: ```suggestion
2. Conditional expressions within double quotes are considered strings.
```

Reviewer 2: And then add the examples suggested by @SudarshanPurohit .

   So, "<+json.select("fields.status.name", httpResponseBody)>"=="In Progress" is treated as string comparison and will not work.

>    Rather use -> <+json.select("fields.status.name", httpResponseBody)>=="In Progress"

Reviewer 2: ```suggestion
   Use `<+json.select("fields.status.name", httpResponseBody)>=="In Progress"` instead.
```

Author: done

>    Same goes for null comparison.

Reviewer 2: ```suggestion
   Similarly, use `<+json.object(httpResponseBody).fields.parent>!=null` for null comparison.
```

Author: done

>    Correct way is -> <+json.object(httpResponseBody).fields.parent>!=null

Reviewer: Tweaking it a bit for readability: 

Any expression passed in double quotes is considered a string and is not evaluated. For example, `"<+json.object(httpResponseBody).fields.parent>"` will be just a string, So, a comparison like `"<+json.select("fields.status.name", httpResponseBody)>"=="In Progress"` is treated as string comparison and will not work as expected. 
Instead, use `<+json.select("fields.status.name", httpResponseBody)>=="In Progress"`. 

The keyword `null`, too, shouldn't be enclosed in quotes during comparison. Here's an example of a null comparison: `<+json.object(httpResponseBody).fields.parent>!=null`

Author: done
